CMakeLists.txt

```diff
@@ -1,571 +1,413 @@
+        $<$<NOT:$<BOOL:${CONFIG_DIR}>>:CONFIG_DIR="${CMAKE_INSTALL_PREFIX}/share/emptyepsilon/">
+)
 
-# Setup OpenGl
-if(ANDROID)
-  target_link_libraries(${EXECUTABLE_NAME} GLESv1_CM)
-else()
-  find_package(OpenGL REQUIRED)
-  include_directories(${OPENGL_INCLUDE_DIR})
-  target_link_libraries(${EXECUTABLE_NAME} ${OPENGL_LIBRARIES})
+# install resources, scripts, and packs to app bundle instead of system dir.
+if(APPLE)
+    set(CMAKE_INSTALL_PREFIX "./")
 endif()
 
-# Opus for voice chat
-add_subdirectory("${SERIOUS_PROTON_DIR}/src/libopus/" "${CMAKE_CURRENT_BINARY_DIR}/libopus")
-target_link_libraries(${EXECUTABLE_NAME} opus)
+target_include_directories(${PROJECT_NAME}
+    PUBLIC
+        $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/src>
+)
 
-set(GLM_BASE_PATH "${CMAKE_CURRENT_BINARY_DIR}/glm")
-set(GLM_ZIP "${GLM_BASE_PATH}/glm.zip")
-if(NOT EXISTS "${GLM_ZIP}")
-  file(DOWNLOAD "https://github.com/g-truc/glm/releases/download/0.9.9.8/glm-0.9.9.8.zip" "${GLM_ZIP}" TIMEOUT 60 TLS_VERIFY ON)
-  execute_process(COMMAND ${CMAKE_COMMAND} -E tar -xf "${GLM_ZIP}" WORKING_DIRECTORY "${GLM_BASE_PATH}")
-endif()
-target_include_directories(${EXECUTABLE_NAME} PUBLIC "${GLM_BASE_PATH}/glm")
+add_library(opengl INTERFACE)
+target_link_libraries(opengl
+  INTERFACE
+  "$<$<BOOL:${ANDROID}>:GLESv1_CM;EGL>"
+  "$<$<NOT:$<BOOL:${ANDROID}>>:OpenGL::GL>"
+)
 
-# Some magical Win 32 stuff
-if(WIN32)
-  target_link_libraries(${EXECUTABLE_NAME} wsock32 iphlpapi)
-  if(ENABLE_CRASH_LOGGER)
-	if(MINGW)
-		include_directories(${DRMINGW_ROOT}/include/)
-	endif()
-    target_link_libraries(${EXECUTABLE_NAME} exchndl)
-  endif()
-  set(DISCORD_ARCH "x86_64")
-  if(${CMAKE_SIZEOF_VOID_P} EQUAL 4)
-	set(DISCORD_ARCH "x86")
-  endif()
-  set(DISCORD_LIB_PATH "${CMAKE_CURRENT_BINARY_DIR}/discord/lib/${DISCORD_ARCH}")
-  if(NOT EXISTS "${DISCORD_LIB_PATH}/discord_game_sdk.dll.lib")
-    file(DOWNLOAD "https://dl-game-sdk.discordapp.net/latest/discord_game_sdk.zip" "${CMAKE_CURRENT_BINARY_DIR}/discord/game_sdk.zip" TIMEOUT 60 TLS_VERIFY ON)
-    execute_process(COMMAND ${CMAKE_COMMAND} -E tar -xf "${CMAKE_CURRENT_BINARY_DIR}/discord/game_sdk.zip" WORKING_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/discord")
-    if(CMAKE_HOST_SYSTEM_NAME STREQUAL "Linux")
-      # Discord uses Windows.h (capital W) and that does not play nice when cross-compiling.
-      # we want to patch with a lowercase windows.h
-      message(STATUS "Patching discord_game._sdk.h, using a lowercase Windows.h (cross-compilation compatibility)")
-      execute_process(COMMAND sed -i "s/<Windows.h>/<windows.h>/g" c/discord_game_sdk.h WORKING_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/discord")
-    endif()
-  endif()
-  
-  target_link_libraries(${EXECUTABLE_NAME} "${DISCORD_LIB_PATH}/discord_game_sdk.dll.lib")
-  target_include_directories(${EXECUTABLE_NAME} PUBLIC "${CMAKE_CURRENT_BINARY_DIR}/discord/c/")
-elseif(ANDROID)
+target_link_libraries(${PROJECT_NAME}
+    PUBLIC
+        seriousproton glm::glm opengl
```

Comment:
I remember having issues with linking seriousproton as library before. Something with static initialization order not going right. But it's a vague memory...

Reply:
You did, but I haven't encountered any issues...

I also use SP as a library for my SDL branch and nothing came up there either.

Currently triple checking (I did Windows, Debian, RPi already but I can do a second pass)

Reply:
Triple checking for the win: I think I saw the issue you recall.

It's not about static initialization order, so much as [LTO bringing mayhem](https://stackoverflow.com/q/27372667) with MinGW.

It required a little finesse, but I have something that's working :) (requires additional tweaks on SP's target setup).